docs/guides/gradle.md

```diff
@@ -0,0 +1,171 @@
+
+You can also find this declaration on the [Gradle Plugin Portal](https://plugins.gradle.org/plugin/io.spine.tools.gradle.bootstrap),
+or on our [Getting Started page]({{site.baseurl}}/docs/quick-start).
+
+## Spine Bootstrap plugin 
+
+Spine Bootstrap plugin (Bootstrap for short) serves to automate the configuration of the modules
+in your Spine-based app.
+
+We recommend having separate Gradle subprojects for domain model definition, server  implementation,
+and client code. Bootstrap is applied a bit differently in each of these cases.
+
+### Model definitions
+
+Let's assume one of the subprojects contains Protobuf definitions for the domain model. This
+subproject is typically called `model`. In `build.gradle` for that subproject, declare:
+```groovy
+spine.assembleModel()
+```
+This way, the subproject will receive all the required Spine dependencies but no code will be
+generated from Protobuf.
+
+### Java server implementation
+
+A separate Gradle subproject is dedicated to the server-side business logic of your Bounded Context.
+Usually, this subproject is named `server` or after the Bounded Context it implements, e.g. `users`.
+In `build.gradle` for that subproject declare:
+```groovy
+spine.enableJava().server()
+
+dependencies {
+    protobuf(project(':model'))
+}
+```
+This will add dependencies to the `spine-server` artifact and set up code generation for
+the domain model types.
+
+It is perfectly normal to have more Protobuf types in these modules, as long as those types are
+internal to your Java implementation and are not a part of the publicly-visible domain model.
+
+For any specific subproject, you can configure to run or skip certain code generation routines.
+For example:
+```groovy
+spine.enableJava {
+    server() // Enable Server API.
+    codegen {
+        protobuf = true // Generate default Protobuf Java classes.
+        spine = false // Avoid enhancing Protobuf Java classes with Spine validation, interfaces, etc.
+        grpc = true // Generate gRPC stubs and implementation bases from Protobuf service definitions.  
+    }
+}
+```
+
+<p class="note">
+Note the use of the `protobuf` configuration. This tells our tools that the Protobuf definitions
+in the subproject `model` must be converted into Java code in the current subproject.
+
+Alternatively, if, for instance, the upstream project already contains code generated from Protobuf,
+and no additional codegen is required, the `api`/`implementation` configurations should be used. See
+[this Gradle doc](https://docs.gradle.org/current/userguide/dependency_management_for_java_projects.html)
+for more info.
+</p>
+
+### Java web server
+
+If your project contains a JavaScript frontend, you may declare a `web-server` subproject, which
+processes the HTTP requests from JS. In `web-server/build.gradle`:
+```groovy
+dependencies {
+    implementation("io.spine:spine-web:${spine.version()}")
+    implenemtation(project(':server'))
```

Comment:
Please fix a typo.

Ideally, we would want to refer to some code in a real project instead. Not sure it exists now, but can we use `todo-list`?